### ISSUE TITLE
Missing values after creating a working question.

### ISSUE THREAD

**Issue by inesmoskal** [metabase/metabase#11495]
**Description**
On some questions, when I try to access them later or create another query on top of them I encounter missing or bugged values. See screenshots. 

**Logs**
```
[085da644-080e-4bee-87e5-ef93c0a39ab5] 2019-12-11T11:24:13+00:00 DEBUG metabase.middleware.log GET /api/user/current 200 12.5 ms (3 DB calls) Jetty threads: 4/50 (3 idle, 0 queued) (59 total active threads) Queries in flight: 0
[085da644-080e-4bee-87e5-ef93c0a39ab5] 2019-12-11T11:24:13+00:00 DEBUG metabase.middleware.log GET /api/session/properties 200 28.3 ms (1 DB calls) Jetty threads: 4/50 (3 idle, 0 queued) (59 total active threads) Queries in flight: 0
[085da644-080e-4bee-87e5-ef93c0a39ab5] 2019-12-11T11:24:13+00:00 DEBUG metabase.middleware.log GET /api/database 200 9.0 ms (3 DB calls) Jetty threads: 3/50 (4 idle, 0 queued) (61 total active threads) Queries in flight: 0
[085da644-080e-4bee-87e5-ef93c0a39ab5] 2019-12-11T11:24:13+00:00 DEBUG metabase.middleware.log GET /api/card/40 200 16.3 ms (6 DB calls) Jetty threads: 4/50 (3 idle, 0 queued) (61 total active threads) Queries in flight: 0
[085da644-080e-4bee-87e5-ef93c0a39ab5] 2019-12-11T11:24:13+00:00 DEBUG metabase.middleware.log GET /api/setting 200 1.5 ms (0 DB calls) Jetty threads: 4/50 (3 idle, 0 queued) (61 total active threads) Queries in flight: 0
[085da644-080e-4bee-87e5-ef93c0a39ab5] 2019-12-11T11:24:13+00:00 DEBUG metabase.middleware.log GET /api/alert/question/40 200 2.5 ms (1 DB calls) Jetty threads: 5/50 (2 idle, 0 queued) (61 total active threads) Queries in flight: 0
[085da644-080e-4bee-87e5-ef93c0a39ab5] 2019-12-11T11:24:13+00:00 DEBUG metabase.middleware.log GET /api/database 200 129.0 ms (7 DB calls) Jetty threads: 3/50 (4 idle, 0 queued) (61 total active threads) Queries in flight: 0
[085da644-080e-4bee-87e5-ef93c0a39ab5] 2019-12-11T11:24:13+00:00 DEBUG metabase.middleware.log GET /api/collection/1 200 6.4 ms (3 DB calls) Jetty threads: 3/50 (4 idle, 0 queued) (61 total active threads) Queries in flight: 0
[085da644-080e-4bee-87e5-ef93c0a39ab5] 2019-12-11T11:24:18+00:00 DEBUG metabase.middleware.log GET /api/session/properties 200 9.0 ms (1 DB calls) Jetty threads: 4/50 (3 idle, 0 queued) (61 total active threads) Queries in flight: 0
[085da644-080e-4bee-87e5-ef93c0a39ab5] 2019-12-11T11:24:18+00:00 DEBUG metabase.middleware.log GET /api/setup/admin_checklist 200 25.3 ms (11 DB calls) Jetty threads: 3/50 (4 idle, 0 queued) (61 total active threads) Queries in flight: 0
[085da644-080e-4bee-87e5-ef93c0a39ab5] 2019-12-11T11:24:18+00:00 DEBUG metabase.middleware.log GET /api/setting 200 695.4 µs (0 DB calls) Jetty threads: 3/50 (4 idle, 0 queued) (61 total active threads) Queries in flight: 0
[085da644-080e-4bee-87e5-ef93c0a39ab5] 2019-12-11T11:24:19+00:00 DEBUG metabase.middleware.log GET /api/util/bug_report_details 200 7.7 ms (1 DB calls) Jetty threads: 3/50 (4 idle, 0 queued) (61 total active threads) Queries in flight: 0
```

**Screenshots**
<img width="1029" alt="missing_values" src="https://user-images.githubusercontent.com/24622098/70617368-5c2d2500-1c08-11ea-84ea-05a82076d560.png">
<img width="980" alt="missing_values_2" src="https://user-images.githubusercontent.com/24622098/70617369-5cc5bb80-1c08-11ea-94c1-0e809aa24363.png">

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.108 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "1.8.0_232-b09",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_232",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "25.232-b09",
    "os.name": "Linux",
    "os.version": "5.1.16-300.fc30.x86_64",
    "user.language": "en",
    "user.timezone": "Etc/UTC"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "tag": "v0.33.6",
      "date": "2019-11-19",
      "branch": "release-0.33.x",
      "hash": "be1e0e1"
    },
    "settings": {
      "report-timezone": null
    }
  }
}

```
**Severity**
It's impossible to achieve some of the charts necessary. 

**Additional context**
It happens on all the fields, there is no rule. So in one case the filter is Unknown, some of the group-by's dissapear, or the method of summarizing.

**Comment by flamber** [comment]
Hi @inesmoskal 
Is `Lines_2_lean` a table or Saved Question? Has the structure of that table/question changed, since you created this question?
Do you see any errors in the browser console?

**Comment by inesmoskal** [comment]
`Lines_2_lean` is a saved question. No, the question underneath didn't change, and I also don't see any errors, it just appears as on the screenshot and nothing wrong in the logs. Sometimes it fixes itself after refreshing, but not always.

**Comment by flamber** [comment]
@inesmoskal Could it be that all the missing fields are from linked tables? Then it sounds like a duplicate of #11235.

**Comment by inesmoskal** [comment]
In the second screenshot `Transactions` it's a table straight from the database. On the first screenshot the `Summarize` field disappeared, and in the second screenshot all of the `Group by` disappeared in the first `Summarize`, and in the second changed to original schema names.

**Comment by flamber** [comment]
@inesmoskal I'm guessing you're working with @EBoisseauSierra and this might be related to #11496.

**Comment by EBoisseauSierra** [comment]
@flamber you're right (; 
The two issue are happening on the same Metabase instance, with the same dataset. I've been trying to replicate the issue with the Sample dataset, without success for the moment.

However, _this issue_ (#11495) arise more frequently than #11496 and on more tables, so my feeling is that this one is I have the feeling

**Comment by flamber** [comment]
@EBoisseauSierra :-)
If you in any way can try to reproduce either of the issues with Sample Dataset, then it would help a lot on debugging this, since there doesn't seem to be any errors (Metabase/browser), so trying to figure out what is going on without access to your data is going to be difficult.
I still have a feeling that it has something to do with #11235, but maybe it's only half of the problem.

**Comment by flamber** [comment]
**Reproduce:**
_Replace Transactions with Orders (warehouse_code => product_id) and Warehouse with Products._
1. First question:
![image](https://user-images.githubusercontent.com/1447303/70653457-28f39180-1c1a-11ea-91f3-4eccb924be57.png)
2. Second question is based on first:
![image](https://user-images.githubusercontent.com/1447303/70653499-34df5380-1c1a-11ea-9024-d5c6f763c24a.png)
3. And then after I have saved the second question, if I refresh the page, then it looses some fields:
![image](https://user-images.githubusercontent.com/1447303/70653901-f1391980-1c1a-11ea-91f6-3cb5e3d5d506.png)
When clicking the empty Summarize field, it errors with the following:
```
Firefox: TypeError: selectedAggregation is undefined
Chromium: Uncaught TypeError: Cannot read property 'name' of undefined
```
While clicking the Group By doesn't do anything, only removing it (`X`) does anything and produces similar errors (card ID of this question is 106 (step 2), while it uses question 105 (step 1) as it's primary data):
```
Firefox:
Request entities,tables,card__105,fetchMetadata failed: TypeError: "selectedAggregation is undefined"
Request entities,tables,card__105,fetchForeignKeys failed: TypeError: "selectedAggregation is undefined"
Error loading metadata for card TypeError: "selectedAggregation is undefined"
Chromium:
Request entities,tables,card__105,fetchMetadata failed: TypeError: Cannot read property 'name' of undefined
Request entities,tables,card__105,fetchForeignKeys failed: TypeError: Cannot read property 'name' of undefined
Error loading metadata for card TypeError: Cannot read property 'name' of undefined
```

**Comment by EBoisseauSierra** [comment]
@flamber thanks for working out a reproducible example! It's indeed the same behaviour (I've been able to replicate it on our servers as well).

FYI, right after creating the first question (the same happend with the second one), I clicked on "Ask a question/Custom question/Saved Question/Étienne Boisseau-Sierra's collection"… and the question I had just created didn't appear. I had to `F5` before it did appear. (Not sure whether it's related, though.)
